docs/guides/src/main/server/hostname.adoc

```diff
@@ -0,0 +1,148 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Configure the Hostname"
+summary="Learn how to use Keycloaks hostname configuration to expose Keycloaks endpoints the right way."
+includedOptions="hostname-* proxy">
+
+When running Keycloak in environments such as Kubernetes, OpenShift or even on-premise, you usually do not want to expose the internal URLs to the public facing internet. Instead, you want to use your public hostname. This guide will show you how to configure Keycloak to use the right hostname for different scenarios.
+
+== Keycloak API Endpoint categories
+Keycloak exposes three different API endpoint categories, each using a specific base URL. These categories are:
+
+=== Frontend Endpoints
+Frontend endpoints are used to externally access Keycloak. When no Hostname is set, the base URL used for the frontend is taken from the incoming request. This has some major disadvantages, e.g. in High availability setups spanning multiple Keycloak instances, where it should not depend on the instance the request lands what URL is used, but instead one URL should be used for all instances, so they are seen as one system from the outside.
+
+The `hostname` property can be used to set the frontend base URL:
+
+<@kc.start parameters="--hostname=<value>"/>
+
+
+=== Backend Endpoints
+Backend endpoints are used for direct communication between Keycloak and applications, so normally without user interaction. Examples are the Token endpoint or the User info endpoint. Backend endpoints are also taking the base URL from the request by default. To override this behaviour, set the `hostname-strict-backchannel` configuration option:
+
+<@kc.start parameters="--hostname=<value> --hostname-strict-backchannel=true"/>
+
+If set to `true`, all endpoints use the URL which was set as `hostname` value.
+
+=== Administrative Endpoints
+Keycloaks administration endpoints and its admin console should usually not be publicly accessible at all. You can force keycloak to use a specific, private URL for all `/admin/` endpoints:
+
+<@kc.start parameters="--hostname=<publicHostname> --hostname-admin=<securedHostname>"/>
+
+When `hostname-admin` is configured, all links and static resources used to e.g. render the admin console are served from the value you enter for `<securedHostname` instead of using `<publicHostname`.
+
+For more information about which paths to expose using a reverse proxy, please refer to the <@links.server id="proxy"/> Guide.
```

Comment:
This is quite misleading as it can be interpreted as this somehow exposing the admin on a different URL. Should make it explicit that this isn't changing how they are exposed, just telling Keycloak where they are exposed.